app/src/main/java/org/mozilla/tv/firefox/ScreenControllerStateMachine.kt

```diff
@@ -0,0 +1,88 @@
+
+// TODO add this stuff
+
+package org.mozilla.tv.firefox
+
+import android.support.annotation.VisibleForTesting
+
+/**
+ * TODO
+ */
+class ScreenControllerStateMachine(private val currentUrlIsHome: () -> Boolean) {
```

Comment:
I wonder if exploring the idea of refactoring all the transitional operations (i.e. Youtube escape) in here worthwhile 🤔

Reply:
Hmm, we could definitely move the "handleYoutube" here, and just call it before calling the browser back behavior, though it still needs the webview in order to call the JS, so the decoupling is limited.